support/Environments/caltech_hpc_gcc.sh

```diff
@@ -0,0 +1,284 @@
+#!/bin/env sh
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
```

Comment:
Could you please add docs on how to build here or in the dev guide? E.g., you have to install the modules, and you have to compile on a compute node. I get out of memory errors when trying to build on the head node, and so far, I only figured out how to get a single core compute node (or messed something else up, so it only builds in serial).